src/interpreter/index.ts

```diff
@@ -252,63 +226,183 @@ function translateStep<T, I>(
+            return interpreteBinaryOperator(value, step, context)
         case "if":
-            return interpreteIf(step, context, next)
+            return interpreteIf(value, step, context)
         case "switch":
-            return interpreteSwitch(step, context, next)
+            return interpreteSwitch(value, step, context)
         case "getVariable":
-            return interpreteGetVariable(step, next)
+            return interpreteGetVariable(value, step)
         case "setVariable":
-            return interpreteSetVariable(step, context, next)
-        case "return":
-            return interpreteReturn()
+            return interpreteSetVariable(value, step, context)
         case "random":
-            return interpreteRandom(step, context, next)
-        case "null":
-            return interpreteNull()
+            return interpreteRandom(value, step, context)
+    }
+    return {} as Value<T>[]
+}
+
+function interpreteRandom<T, A, I>(
+    value: Value<T>[],
+    step: AbstractParsedRandom<I>,
+    context: InterpretionContext<T, I>
+): Value<T>[] {
+    const newValue = [] as Value<T>[]
+
+    const stepProb: number[] = []
+    step.probabilities.forEach((curr, i) =>
+        i == 0 ? stepProb.push(curr) : stepProb.push(curr + stepProb[stepProb.length - 1])
+    )
+    value.forEach((v) => {
+        const rand = v3(v.index.join(","), context.seed) / _32bit_max_int
+        //const rand = Math.random()
+        const stepIndex = stepProb.findIndex((e) => rand <= e)
+        if (stepIndex !== -1) {
+            if (context.listeners?.onRandom != null) {
+                const onRandom = context.listeners.onRandom
+                onRandom(step, v, stepIndex)
+            }
+            const interpreVals = interpreteStep([v], step.children[stepIndex], context)
+            newValue.push(...interpreVals)
+        }
+    })
+    return newValue
+}
+
+function interpreteGetVariable<T>(value: Value<T>[], step: ParsedGetVariable): Value<T>[] {
+    return value[0].variables[step.identifier]
+}
+
+function interpreteSetVariable<T, I>(
+    value: Value<T>[],
+    step: ParsedSetVariable,
+    context: InterpretionContext<T, I>
+): Value<T>[] {
+    const valueOperatorFunction = interpreteStep(value, step.children[0], context)
+    value.map((v) => {
+        return { ...v, variables: { ...v.variables, [step.identifier]: valueOperatorFunction } }
+    })
+    return value
+}
+
+function interpreteSwitch<T, I>(value: Value<T>[], step: ParsedSwitch, context: InterpretionContext<T, I>): Value<T>[] {
+    //const valueOperator = interpreteStep(value, step.children[0], context)[0]
+    //const casesOperatorFunctions = step.children.slice(1).map((child) => interpreteStep(child, context, next))
+    let newValue = value
+    const conditionOperatorValue = interpreteStep(value, step.children[0], context)[0]
+    step.cases.forEach((v, i) => {
+        if (v.some((e) => e === conditionOperatorValue.raw)) {
```

Comment:
problem ist hier das du beim 1. match nicht abbrichst und weiter nach matches suchst. Das heißt das ergebnis ist immer das vom letzen match. Ich würde eine `for(const case of cases)` nutzen und manuell einen index hochzählen. Bei einem match direkt `return interpreteStep(...)` und bei keinem match einen leeren array.

Aktuell gibst du bei keinem match den input wert zurück, was ich nicht korrekt finde.